arraycontext/container/traversal.py

```diff
@@ -522,4 +526,44 @@ def to_numpy(ary: Any, actx: ArrayContext) -> Any:
 
 # }}}
 
+
+# {{{ algebraic operations
+
+def outer(a: Any, b: Any) -> Any:
+    """
+    Compute the outer product of *a* and *b*.
+
+    Tweaks the behavior of :func:`numpy.outer` to return a lower-dimensional
+    object if either/both of *a* and *b* are scalars (whereas :func:`numpy.outer`
+    always returns a matrix). Here the definition of "scalar" includes
+    all non-array-container types and any scalar-like array container types.
+
+    If *a* and *b* are both array containers, the result will have the same type
+    as *a*. If both are array containers and neither is an object array, they must
+    have the same type.
+    """
+
+    def is_scalar(x: Any) -> bool:
+        if is_array_container(x):
+            return (
+                not isinstance(x, np.ndarray)
+                and np.ndarray not in x.__class__._outer_bcast_types)
+        else:
+            return True
+
+    if is_scalar(a) or is_scalar(b):
+        return a*b
+    elif isinstance(a, np.ndarray) and isinstance(b, np.ndarray):
```

Comment:
Narrow to only object arrays in view of #95

Reply:
Fixed. (3c9da3a)